File: jhyun/Dijkstra/각_정점까지의_최단_경로3.java
import java.util.*;
import java.io.*;

public class Main {
    static final int MAX_N = 100;
    static int n, m;
    static int[][] graph = new int[MAX_N + 1][MAX_N + 1];
    static boolean[] visited = new boolean[MAX_N + 1];
    static int[] dist = new int[MAX_N + 1];

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());

        for (int i = 0; i < m; i++) {
            st = new StringTokenizer(br.readLine());
            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());
            int z = Integer.parseInt(st.nextToken());
            graph[x][y] = z;
        }

        //그래프에 있는 모든 노드들에 대해 초기값을 전부 아주 큰 값으로 설정
        //INT_MAX 그 자체로 설정하면 후에 덧셈 진행시 overflow가 발생할 수도 있으니
        //적당히 큰 숫자로 적어줘아함!
        for (int i = 1; i <= n; i++) {
            dist[i] = (int) 1e9;
        }

        //시작위치에는 dist값을  0으로 설정
        dist[1] = 0;

        //O(|V|^2) 다익스트라 코드
        for (int i = 1; i <= n; i++) {
            //V개의 정점 중 아직 방문하지 않은 정점 중
            //dist값이 가장 작은 정점을 찾아줌
            int minIndex = -1;
            for (int j = 1; j <= n; j++) {
                if (visited[j]) {
                    continue;
                }
                if (minIndex == -1 || dist[minIndex] > dist[j]) {
                    minIndex = j;
                }
            }

            //최솟값에 해당하는 정점에 방문 표시 진행
            visited[minIndex] = true;

            //최솟값에 해당하는 정점에 연결된 간선들을 보며 시작점으로부터의 최단거리 값을 갱신
            for (int j = 1; j <= n; j++) {
                //간선이 존재하지 않는 경우에는 넘어감
                if (graph[minIndex][j] == 0) {
                    continue;
                }
                dist[j] = Math.min(dist[j], dist[minIndex] + graph[minIndex][j]);
            }
        }

        //시작점(1번 정점)으로부터 각 지점까지의 최단거리 값을 출력
        for (int i = 2; i <= n; i++) {
            //만약 도달이 불가능하다면 -1
            if (dist[i] == (int) 1e9) {
                System.out.println(-1);
            } else {
                System.out.println(dist[i]);
            }
        }
    }
}

File: Park/CodeTree/Shortest Path/다익스트라/각 정점까지의 최단경로3.py
import sys

INT_MAX = sys.maxsize  # 아주 큰 값을 나타냄.

# 변수 선언 및 입력
n, m = map(int, input().split())  # n은 정점의 수, m은 간선의 수
graph = [[0] * (n + 1) for _ in range(n + 1)]  # 인접 행렬로 그래프 표현
visited = [False] * (n + 1)  # 각 정점의 방문 여부를 저장

# 각 정점까지의 최단거리를 나타내는 배열을 아주 큰 값으로 초기화
dist = [INT_MAX] * (n + 1)  

# 그래프를 인접 행렬로 입력 받기.
for _ in range(m):
    x, y, z = map(int, input().split())  # x에서 y로 가는 가중치 z의 간선
    graph[x][y] = z  # 간선을 인접 행렬에 기록

# 시작점(1번 정점)의 거리를 0으로 설정
dist[1] = 0

# 다익스트라 알고리즘 구현
for i in range(1, n + 1):
    # 방문하지 않은 정점 중 최단 거리가 가장 작은 정점을 찾기.
    min_index = -1
    for j in range(1, n + 1):
        if not visited[j] and (min_index == -1 or dist[min_index] > dist[j]):
            min_index = j

    # 해당 정점을 방문 처리
    visited[min_index] = True

    # 선택된 정점과 연결된 다른 정점들의 거리 갱신
    for j in range(1, n + 1):
        if graph[min_index][j] != 0:  # 간선이 존재할 때만
            dist[j] = min(dist[j], dist[min_index] + graph[min_index][j])

# 결과 출력: 1번 정점에서 각 정점까지의 최단거리를 출력.
for i in range(2, n + 1):
    if dist[i] == INT_MAX:
        print(-1)  # 도달할 수 없는 경우
    else:
        print(dist[i])  # 최단 거리 출력
